aapcs64-morello/aapcs64-morello.rst

```diff
@@ -634,25 +668,28 @@ The differences in language bindings used for AAPCS64 and AAPCS64-cap are descri
     +-------------------------------+----------------------------------------------------------------------------------------+
     |                               | The NSAA is rounded up to the larger of 8 or the Natural Alignment of the argument's   |
     |                               | type.                                                                                  |
-    | C.13                          |                                                                                        |
+    | C.15                          |                                                                                        |
     +-------------------------------+----------------------------------------------------------------------------------------+
     |                               | If the argument is a composite type then the argument is copied to memory at the       |
     |                               | adjusted NSAA. The NSAA is incremented by the size of the argument. The argument has   |
-    | C.14                          | now been allocated.                                                                    |
+    | C.16                          | now been allocated.                                                                    |
     +-------------------------------+----------------------------------------------------------------------------------------+
     |                               | If the size of the argument is less than 8 bytes then the size of the argument is set  |
     |                               | to 8 bytes. The effect is as if the argument was copied to the least significant bits  |
-    | C.15                          | of a 64-bit register and the remaining bits filled with unspecified values.            |
+    | C.17                          | of a 64-bit register and the remaining bits filled with unspecified values.            |
     +-------------------------------+----------------------------------------------------------------------------------------+
     |                               | The argument is copied to memory at the adjusted NSAA.  The NSAA is incremented by the |
     |                               | size of the argument. The argument has now been allocated.                             |
-    | C.16                          |                                                                                        |
+    | C.18                          |                                                                                        |
     +-------------------------------+----------------------------------------------------------------------------------------+
 
+.. note::
+
+    In AAPCS64-cap if the callee is variadic and there are less than 4094 Anonymous arguments, the length of C9 divided by 16 is equal to the number of Anonymous arguments. The length of C9 divided by 16 is always greater or equal to the number of Anonymous arguments.
```

Comment:
Nitpick, but  s/less/fewer/ for the number of arguments.

Reply:
This is now removed. The syscall interface shouldn't be relying on this in the first place, and instead it should be getting the number of arguments by doing (length-offset)/16.

Reply:
OK that sounds like a good approach, but I feel that having a note is still useful, otherwise there's no guarantee that the kernel can rely on the bounds being exact.

Reply:
Discussed offline, this is not actually needed - even if there is zero padding at the end, loading additional zeroes is harmless.

Reply:
Re-added this after some more offline discussions. I also did some experiments, and we always get exact bounds for fewer than 4096 arguments (also adding more arguments means that we don't) .